src/main/java/svc/types/HashableEntity.java

```diff
@@ -0,0 +1,24 @@
+package svc.types;
+
+public class HashableEntity<CLASS>{
+	public Class<CLASS> type;
+	
+	public Long value;
+	
+	public HashableEntity(Class<CLASS> cls, Long id){
+		this.value = id;
+		this.type = cls;
+	}
+	
+	public Long getValue(){
```

Comment:
You don't need BOTH public properties AND getters/setters. Just use one or the other